stl/inc/xmemory

```diff
@@ -2068,6 +2072,105 @@ void _Erase_nodes_if(_Container& _Cont, _Pr _Pred) { // erase each element satis
         }
     }
 }
+
+#if _HAS_CXX20
+
+// VARIABLE TEMPLATE _Is_pair_v
+template <class>
+inline constexpr bool _Is_pair_v = false;
+
+template <class _Ty1, class _Ty2>
+inline constexpr bool _Is_pair_v<pair<_Ty1, _Ty2>> = true;
+
+template <class _Ty, class _Alloc, class... _Types, enable_if_t<!_Is_pair_v<_Ty>, int> = 0>

```

Comment:
It's a c++20 feature. Can't we use concepts there instead of SFINAE?

Reply:
Not sure if it is allowed, in standard template parameters are plain `class`es. Plus we'd need 2 concepts, Pair and NotPair. Anyway, I think this sfinae is rather simple, but I'd rewrite it if concepts are faster in terms of compilation speed

Reply:
We don't need to make any concepts we can just use `requires` keyword.

Reply:
Moreover standard uses word _Constraints_

> _Constraints: T is not a specialization of pair._

which evidently suggests the use of concepts.

Reply:
I think concepts are generally preferable in newer code as they are simpler and provide better error messages. 
Nevertheless whether you use a concept or SFINAE is your choice.

Reply:
SFINAE is horrible. Working with it is one of my biggest nightmares.

Reply:
> It's a c++20 feature. Can't we use concepts there instead of SFINAE?

Sadly not all the frontends we support have concepts yet.

> SFINAE is horrible. Working with it is one of my biggest nightmares.

concepts are mostly SFINAE :)

Reply:
Fair. I'll keep `_Is_pair` and replace `enable_if`s with requires

Reply:
> Sadly not all the frontends we support have concepts yet.

Yet. Someday they will.

> concepts are mostly SFINAE :)

But 100x simpler and easier to debug.

Reply:
> Sadly not all the frontends we support have concepts yet.

So enable_if remains for now?